src/Exec_Top.cpp

```diff
@@ -19,105 +20,150 @@ Exec::RetType Exec_ParmInfo::Execute(CpptrajState& State, ArgList& argIn) {
   parm->Summary();
   return CpptrajState::OK;
 }
+
+// -----------------------------------------------------------------------------
+static int CommonSetup(TopInfo& info, CpptrajState& State, ArgList& argIn, const char* desc)
+{
+  Topology* parm = 0;
+  ReferenceFrame REF = State.DSL().GetReferenceFrame( argIn );
+  if (REF.error()) return 1;
+  if (REF.empty()) {
+    parm = State.DSL().GetTopByIndex( argIn );
+    if (parm == 0) return 1;
+  } else
+    mprintf("\tUsing '%s'\n", REF.refName());
+  std::string outname = argIn.GetStringKey("out");
+  int err = 0;
+  if (outname.empty())
+    err = info.SetupTopInfo( parm, REF.RefPtr() );
+  else {
+    CpptrajFile* outfile = State.DFL().AddCpptrajFile(outname, desc);
+    if (outfile == 0) return CpptrajState::ERR;
+    mprintf("\tOutput to '%s'\n", outfile->Filename().full());
+    err = info.SetupTopInfo( outfile, parm, REF.RefPtr() );
+  }
+  return err;
+}
+
 // -----------------------------------------------------------------------------
 void Exec_BondInfo::Help() const {
-  mprintf("\t[%s] [<mask>]\n", DataSetList::TopIdxArgs);
-  mprintf("  Print bond info for atoms in <mask> for specified topology (first by default).\n");
+  mprintf("\t[%s] [<mask1>] [<mask2>] [out <file>]\n", DataSetList::TopIdxArgs);
+  mprintf("  For specified topology (first by default) either print bond info for all\n"
+          "  atoms in <mask1>, or print info for bonds with first atom in <mask1> and\n"
+          "  second atom in <mask2>.\n");
 }
 
 Exec::RetType Exec_BondInfo::Execute(CpptrajState& State, ArgList& argIn) {
-  Topology* parm = State.DSL().GetTopByIndex( argIn );
-  if (parm == 0) return CpptrajState::ERR;
-  parm->PrintBondInfo( argIn.GetMaskNext() );
+  TopInfo info;
+  if (CommonSetup(info, State, argIn, "Bond info")) return CpptrajState::ERR;
+  std::string mask1 = argIn.GetMaskNext();
+  if (info.PrintBondInfo( mask1, argIn.GetMaskNext() )) return CpptrajState::ERR;
   return CpptrajState::OK;
 }
 // -----------------------------------------------------------------------------
 void Exec_AngleInfo::Help() const {
-  mprintf("\t[%s] [<mask>]\n", DataSetList::TopIdxArgs);
-  mprintf("  Print angle info for atoms in <mask> for specified topology (first by default).\n");
+  mprintf("\t[%s] [<mask1>] [<mask2> <mask3>]\n\t[out <file>]\n", DataSetList::TopIdxArgs);
```

Comment:
In ParmEd, mask2 and mask3 are independently optional.  If you specify 2 masks, the middle atom needs to be in mask2 and either of the other atoms can be in mask1.

It's a little more logic checking, but still useful sometimes.